hyp3lib/get_orb.py

```diff
@@ -1,197 +1,140 @@
 """Get Sentinel-1 orbit file(s) from ASF or ESA website"""
 
-from __future__ import print_function, absolute_import, division, unicode_literals
-
-import re
-from lxml import html
+import argparse
+import logging
 import os
+import re
+import sys
 from datetime import datetime, timedelta
-import argparse
-from hyp3lib.verify_opod import verify_opod
+
 import requests
-import json
+from lxml import html
 from requests.adapters import HTTPAdapter
 from six.moves.urllib.parse import urlparse
 
+from hyp3lib import OrbitDownloadError
+from hyp3lib.fetch import download_file
+from hyp3lib.verify_opod import verify_opod
 
-class FileException(Exception):
-  """Could not download orbit file"""
 
+def _get_asf_orbit_url(search_url, platform, timestamp, verify=True):
+    if not search_url.endswith('/'):
+        search_url += '/'
 
-def getPageContentsESA(url, verify):
-    print("Getting result of {}".format(url))
-    hostname = urlparse(url).hostname
+    hostname = urlparse(search_url).hostname
     session = requests.Session()
     session.mount(hostname, HTTPAdapter(max_retries=10))
-    page = session.get(url, timeout=60, verify=verify)
-    print(page)
-    results = json.loads(page.text)
-    print("results : {}".format(results))
-    if results['count'] > 0:
-        print(results['results'][0]['physical_name'])
-        fileName = results['results'][0]['physical_name']
-        url = results['results'][0]['remote_url']
-        return(fileName,url)
-    else:
-        print("WARNING: No results returned from ESA query")
-    return(None,None)
-
-def getPageContents(url, verify):
-    hostname = urlparse(url).hostname
-    session = requests.Session()
-    session.mount(hostname, HTTPAdapter(max_retries=10))
-    page = session.get(url, timeout=60, verify=verify)
+    page = session.get(search_url, timeout=60, verify=verify)
     tree = html.fromstring(page.content)
-    l = tree.xpath('//a[@href]//@href')
-    ret = []
-    for item in l:
+    file_list = []
+    for item in tree.xpath('//a[@href]//@href'):
         if 'EOF' in item:
-            ret.append(item)
-    return ret
-
-def dateStr2dateTime(string):
-    year = string[0:4]
-    month = string[4:6]
-    day = string[6:8]
-    hour = string[8:10]
-    minute = string[10:12]
-    second = string[12:14]
-    outTime = datetime.strptime("{}-{}-{}T{}:{}:{}".format(year,month,day,hour,minute,second),'%Y-%m-%dT%H:%M:%S')
-    return(outTime)
-
-def findOrbFile(plat,tm,lst):
+            file_list.append(item)
+
```

Comment:
is a list comprehension better?  also `items` here are full urls rather than filenames
`orbit_urls = [url for url in tree.xpath('//a[@href]//@href') if url.endswith('.EOF')]`
or
```
urls = tree.xpath('//a[@href]//@href')
orbit_urls = [url for url in urls if url.endswith('.EOF')]
```

Reply:
They are actually file names: 

```py
orbit_type = 'AUX_POEORB'
search_url = f'https://s1qc.asf.alaska.edu/{orbit_type.lower()}/'
granule = 'S1A_OPER_AUX_POEORB_OPOD_20150711T121908_V20150620T225944_20150622T005944'
platform = platform = granule[0:3]
time_stamps = re.split('_+', granule)[4:6]
search_url = f'https://s1qc.asf.alaska.edu/{orbit_type.lower()}/'
... 
... hostname = urlparse(search_url).hostname
... session = requests.Session()
... retries = Retry(
...     total=3,
...     backoff_factor=10,
...     status_forcelist=[429, 500, 503, 504],
... )
... session.mount(hostname, HTTPAdapter(max_retries=retries))
... response = session.get(search_url)
... response.raise_for_status()
... tree = html.fromstring(response.content)
... file_list = [file for file in tree.xpath('//a[@href]//@href') if file.endswith('.EOF')]
file_list
['S1A_OPER_AUX_POEORB_OPOD_20140822T122852_V20140731T225944_20140802T005944.EOF', ...
```

but I do like list comprehension better here